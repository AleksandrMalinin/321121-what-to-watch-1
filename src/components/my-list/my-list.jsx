import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreators, Operation} from '../../reducer/data/data.js';
import {getFavouriteFilms} from '../../reducer/data/selectors.js';
import withPrivateRoute from '../../hocs/with-private-route/with-private-route.js';
import Sprite from '../sprite/sprite.jsx';
import Logo from '../logo/logo.jsx';
import UserBlock from '../user-block/user-block.jsx';
import MovieList from '../movie-list/movie-list.jsx';

class MyList extends PureComponent {
  render() {
    const {moviesFavourite, handleGenreChange} = this.props;

    return <React.Fragment>
      <Sprite/>

      <div className="user-page">
        <header className="page-header user-page__head">
          <Logo/>
          <UserBlock/>
        </header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <MovieList
            movies={moviesFavourite}
            handleChange={handleGenreChange}
          />

        </section>
        <footer className="page-footer">
          <div className="logo">
            <a href="main.html" className="logo__link logo__link--light">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="copyright">
            <p>Â© 2019 What to watch Ltd.</p>
          </div>
        </footer>
      </div>
    </React.Fragment>;
  }

  componentDidMount() {
    this.props.handleFavouriteFilmsLoad();
  }
}

MyList.propTypes = {
  moviesFavourite: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    genre: PropTypes.string
  })).isRequired,
  handleGenreChange: PropTypes.func,
  handleFavouriteFilmsLoad: PropTypes.func
};

const mapStateToProps = (state) => ({
  moviesFavourite: getFavouriteFilms(state)
});

const mapDispatchToProps = (dispatch) => ({
  handleGenreChange: (genre) => {
    dispatch(ActionCreators.changeGenre(genre));
  },

  handleFavouriteFilmsLoad: () => {
    dispatch(Operation.loadFavouriteFilms());
  }
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(withPrivateRoute(MyList));
