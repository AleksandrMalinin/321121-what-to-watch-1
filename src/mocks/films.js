/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    title: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    title: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    title: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    title: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];
