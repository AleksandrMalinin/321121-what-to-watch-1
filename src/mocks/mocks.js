/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];

export const defaultMovie = {
  background_color: `#AD9F8B`,
  background_image: `https://es31-server.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`,
  description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
  director: `Wes Anderson`,
  genre: `Adventure`,
  id: 1,
  is_favorite: false,
  name: `Dardjeeling Limited`,
  poster_image: `https://es31-server.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  preview_image: `https://es31-server.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`,
  preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.6,
  released: 2007,
  run_time: 91,
  scores_count: 165106,
  starring: [`Owen Wilson`, `Adrien Brody`, `Jason Schwartzman`],
  video_link: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`
};
