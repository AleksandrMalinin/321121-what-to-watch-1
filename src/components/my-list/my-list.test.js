import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {MyList} from './my-list.jsx';
import {films} from '../../mocks/mocks.js';

it(`MyList correctly renders`, () => {
  const renderer = new ShallowRenderer();
  const mockFunction = jest.fn();
  const tree = renderer
  .render(
      <MyList
        moviesFavourite={films}
        onGenreChange={mockFunction}
      />
  );

  expect(tree).toMatchSnapshot();
});
