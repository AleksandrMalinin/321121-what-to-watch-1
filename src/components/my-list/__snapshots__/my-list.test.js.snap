// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList correctly renders 1`] = `
<React.Fragment>
  <Sprite />
  <div
    className="user-page"
  >
    <header
      className="page-header user-page__head"
    >
      <Logo />
      <Memo(ConnectFunction) />
    </header>
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <WithActiveItem
        handleChange={[MockFunction]}
        movies={
          Array [
            Object {
              "genre": "Comedies",
              "name": "Fantastic Beasts: The Crimes of Grindelwald",
              "preview_image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "preview_video_link": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
            Object {
              "genre": "Crime",
              "name": "Bohemian Rhapsody",
              "preview_image": "img/bohemian-rhapsody.jpg",
              "preview_video_link": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
            Object {
              "genre": "Documentary",
              "name": "Macbeth",
              "preview_image": "img/macbeth.jpg",
              "preview_video_link": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
            Object {
              "genre": "Dramas",
              "name": "Aviator",
              "preview_image": "img/aviator.jpg",
              "preview_video_link": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
            Object {
              "genre": "Horror",
              "name": "We need to talk about Kevin",
              "preview_image": "img/we-need-to-talk-about-kevin.jpg",
              "preview_video_link": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
          ]
        }
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</React.Fragment>
`;
